### PR TITLE
User Model: validations, photo, secure password storage

### PR DESCRIPTION
**Pull request by IvanKhudyakov**
@JaneGH @ElijahMSmith the task 3-user-model is ready for review and merge

### CODE REVIEW

File: src/models/User.ts [hunk 1]
```diff
@@ -1,31 +1,68 @@
 import mongoose, { Document, Schema } from "mongoose";
+import bcrypt from "bcryptjs";
+
+const root = process.env.PROFILE_PICTURE_ROOT;
 
-// Define User Schema
 const userSchema = new Schema(
   {
     name: {
```
Comment (JaneGH) [inline src/models/User.ts:8]: Fields should be named like in our file

File: src/models/User.ts [hunk 2]
```diff
@@ -1,31 +1,68 @@
...
+      ],
       unique: true,
     },
     password: {
       type: String,
-      required: true,
+      required: [true, 'Please provide the password'],
+      minLength: 6,
+    },
+    photo: {
```
Comment (ElijahMSmith) [inline src/models/User.ts:33]: Is this really going to work? Presumably we are passing this photo back to the frontend to be displayed, but they won't know what `./images/default-avatar.jpg` is or how to use it.

If we are really interested in profile photos, we'll need some way to store images in our backend and pass them to the frontend. One way of doing this that I've heard of before is "blob storage" (https://softwareengineering.stackexchange.com/questions/296985/where-is-the-best-place-to-store-users-profile-pictures), but I'm sure that any cloud provider has ways of storing photos as well.

I don't think we should worry about this for now. Maybe it's a stretch goal if we have time, but for now there are more important things to be done!

File: src/models/User.ts [hunk 3]
```diff
@@ -1,31 +1,68 @@
 import mongoose, { Document, Schema } from "mongoose";
+import bcrypt from "bcryptjs";
+
+const root = process.env.PROFILE_PICTURE_ROOT;
```
Comment (ElijahMSmith) [inline src/models/User.ts:4]: We should indicate on the README what this variable should be. I would possibly argue that if this path is the same for everyone it shouldn't be a variable, but if this is going to be different when we build and deploy this app then it still makes sense to be a variable.

**Review by ElijahMSmith — APPROVED** [review]

File: src/models/User.ts (hunk 2 above)
Comment (IvanKhudyakov) [reply src/models/User.ts:33]: > Is this really going to work? Presumably we are passing this photo back to the frontend to be displayed, but they won't know what `./images/default-avatar.jpg` is or how to use it.
> 
> If we are really interested in profile photos, we'll need some way to store images in our backend and pass them to the frontend. One way of doing this that I've heard of before is "blob storage" (https://softwareengineering.stackexchange.com/questions/296985/where-is-the-best-place-to-store-users-profile-pictures), but I'm sure that any cloud provider has ways of storing photos as well.
> 
> I don't think we should worry about this for now. Maybe it's a stretch goal if we have time, but for now there are more important things to be done!

Thanks for the comment and link. I used the mongoose help as a starting point. I saw as well some cloud storage and read that storing on the backend might cause some consequences (including security). I would like to make the API work and then test how it works. Based on this, we can adjust the code later.

**Comment by JaneGH** [comment]
don't need anymore

**Pull request closed without merging**